samples/src/main/python/SynapseSample.py

```diff
@@ -0,0 +1,122 @@
+    .option("kustoTable", "<Table name>") \
+    .mode("Append") \
+    .save()
+
+
+#######################################################
+# BATCH WRITE EXAMPLE WITH ADVANCED INGESTION OPTIONS #
+#######################################################
+
+# Please refer to https://github.com/Azure/azure-kusto-spark/blob/master/connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoSinkOptions.scala
+# to get the string representation of the options you need
+
+extentsCreationTime = sc._jvm.org.joda.time.DateTime.now().plusDays(1)
+csvMap = "[{\"Name\":\"ColA\",\"Ordinal\":0},{\"Name\":\"ColB\",\"Ordinal\":1}]"
+# Alternatively use an existing csv mapping configured on the table and pass it as the last parameter of SparkIngestionProperties or use none
+
+
+sp = sc._jvm.com.microsoft.kusto.spark.datasink.SparkIngestionProperties(
+        False, ["dropByTags"], ["ingestByTags"], ["tags"], ["ingestIfNotExistsTags"], extentsCreationTime, csvMap, None)
+# Class fields: SparkIngestionProperties(flushImmediately: Boolean,
+#                                        dropByTags: util.ArrayList[String],
+#                                        ingestByTags: util.ArrayList[String],
+#                                        additionalTags: util.ArrayList[String],
+#                                        ingestIfNotExists: util.ArrayList[String],
+#                                        creationTime: DateTime,
+#                                        csvMapping: String,
+#                                        csvMappingNameReference: String)
+#
+# More info on Ingestion Properties: https://docs.microsoft.com/en-us/azure/data-explorer/ingestion-properties
+
+
+df.write \
+    .format("com.microsoft.kusto.spark.synapse.datasource") \
+    .option("spark.synapse.linkedService", "<link service name>") \
+    .option("kustoDatabase", "<Database name>") \
+    .option("kustoTable", "<Table name>") \
+    .option("sparkIngestionPropertiesJson", sp.toString()) \
+    .option("tableCreateOptions","CreateIfNotExist") \
+    .mode("Append") \
+    .save()
+
+
+
+############################################################
+# STREAMING SPARK DATAFRAME TO A AZURE DATA EXPLORER TABLE #
+############################################################
+
+# Write a streaming Spark DataFrame to a Azure Data Explorer table
+
+
+dfStream = spark \
+  .readStream \
+  .schema([customSchema]) \
+  .csv([filename]) \
+
+spark.conf.set("spark.sql.streaming.checkpointLocation", "/localWriteCheckpointFolder")
+spark.conf.set("spark.sql.codegen.wholeStage", "false")
+
+kustoQ = dfStream \
+    .writeStream \
+    .format("com.microsoft.kusto.spark.synapse.datasink.KustoSynapseSinkProvider") \
+    .option("spark.synapse.linkedService", "<link service name>") \
+    .option("kustoDatabase", "<Database name>") \
+    .option("kustoTable", "<Table name>") \
+    .trigger(once = True)
+    
+kustoQ.start().awaitTermination(60*8)
```

Comment:
i think its better here to not have an await time - as usually when working from databricks/synapse we dont won't to stop the streaming job

Reply:
As discussed, please update as you think fit.